[Blazor] Microsoft.AspNetCore.Components.WebView.props is not imported by Maui apps

We need to figure out why this is not being imported by Maui apps

Hi,
I developed the web app in Blazor with web assembly this is working perfectly fine.
then I add another project "Blazor Hybrid MAUI" to my Solution.
I set up the project according to the documentation but Android opened Splash and home screen then webview Crash the app

Is any additional code I need to integrate in the MAUI.cs file ?

In the following log, I am showing where my app crashed on emulator API level 31.

10:36:52:022 > am start -a "android.intent.action.MAIN" -c "android.intent.category.LAUNCHER" -n "com.companyname.remittance.mobile.ui/crc64a20e96d7d12de042.MainActivity"
10:36:52:022 > Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=com.companyname.remittance.mobile.ui/crc64a20e96d7d12de042.MainActivity }
10:36:52:100 [tance.mobile.ui] Late-enabling -Xcheck:jni
10:36:52:100 [tance.mobile.ui] Unexpected CPU variant for x86: x86_64.
10:36:52:100 [tance.mobile.ui] Known variants: atom, sandybridge, silvermont, kabylake, default
10:36:52:100 [CompatibilityChangeReporter] Compat change id reported: 171979766; UID 10164; state: ENABLED
10:36:52:100 [nativeloader] Configuring classloader-namespace for other apk /system_ext/framework/androidx.window.extensions.jar. target_sdk_version=31, uses_libraries=ALL, library_path=/data/app/~~OIyIRcoljH0lCD6r-U00Yg==/com.companyname.remittance.mobile.ui-Jec8b_LH6cS7DJdCzStj4w==/lib/x86_64:/data/app/~~OIyIRcoljH0lCD6r-U00Yg==/com.companyname.remittance.mobile.ui-Jec8b_LH6cS7DJdCzStj4w==/base.apk!/lib/x86_64, permitted_path=/data:/mnt/expand:/data/user/0/com.companyname.remittance.mobile.ui
10:36:52:100 [nativeloader] Configuring classloader-namespace for other apk /system_ext/framework/androidx.window.sidecar.jar. target_sdk_version=31, uses_libraries=ALL, library_path=/data/app/~~OIyIRcoljH0lCD6r-U00Yg==/com.companyname.remittance.mobile.ui-Jec8b_LH6cS7DJdCzStj4w==/lib/x86_64:/data/app/~~OIyIRcoljH0lCD6r-U00Yg==/com.companyname.remittance.mobile.ui-Jec8b_LH6cS7DJdCzStj4w==/base.apk!/lib/x86_64, permitted_path=/data:/mnt/expand:/data/user/0/com.companyname.remittance.mobile.ui
10:36:52:100 [ziparchive] Unable to open '/data/app/~~OIyIRcoljH0lCD6r-U00Yg==/com.companyname.remittance.mobile.ui-Jec8b_LH6cS7DJdCzStj4w==/base.dm': No such file or directory
10:36:52:100 [ziparchive] Unable to open '/data/app/~~OIyIRcoljH0lCD6r-U00Yg==/com.companyname.remittance.mobile.ui-Jec8b_LH6cS7DJdCzStj4w==/base.dm': No such file or directory
10:36:52:101 [nativeloader] Configuring classloader-namespace for other apk /data/app/~~OIyIRcoljH0lCD6r-U00Yg==/com.companyname.remittance.mobile.ui-Jec8b_LH6cS7DJdCzStj4w==/base.apk. target_sdk_version=31, uses_libraries=, library_path=/data/app/~~OIyIRcoljH0lCD6r-U00Yg==/com.companyname.remittance.mobile.ui-Jec8b_LH6cS7DJdCzStj4w==/lib/x86_64:/data/app/~~OIyIRcoljH0lCD6r-U00Yg==/com.companyname.remittance.mobile.ui-Jec8b_LH6cS7DJdCzStj4w==/base.apk!/lib/x86_64, permitted_path=/data:/mnt/expand:/data/user/0/com.companyname.remittance.mobile.ui
10:36:52:101 [GraphicsEnvironment] ANGLE Developer option for 'com.companyname.remittance.mobile.ui' set to: 'default'
10:36:52:101 [GraphicsEnvironment] ANGLE GameManagerService for com.companyname.remittance.mobile.ui: false
10:36:52:101 [GraphicsEnvironment] Neither updatable production driver nor prerelease driver is supported.
10:36:52:101 [NetworkSecurityConfig] No Network Security Config specified, using platform default
10:36:52:101 [NetworkSecurityConfig] No Network Security Config specified, using platform default
10:36:52:101 [DOTNET] JNI_OnLoad: JNI_OnLoad in pal_jni.c
10:36:52:101 [DOTNET] GetOptionalClassGRef: optional class com/android/org/conscrypt/OpenSSLEngineImpl was not found
10:36:52:101 [monodroid] Creating public update directory: /data/user/0/com.companyname.remittance.mobile.ui/files/.__override__
10:36:52:101 [tance.mobile.ui] Attempt to remove non-JNI local reference, dumping thread
10:36:52:101 [monodroid-debug] Trying to initialize the debugger with options: --debugger-agent=transport=dt_socket,loglevel=0,address=10.0.2.2:59449,embedding=1
10:36:52:910 [monodroid-gc] GREF GC Threshold: 46080
10:36:52:910 [tance.mobile.ui] Symbol file data-0x7c0d2596f050 doesn't match image Mono.Android.dll
10:36:52:910 Loaded assembly: Remittance.Mobile.UI.dll
10:36:52:910 Loaded assembly: Mono.Android.dll [External]
10:36:52:910 Loaded assembly: System.Runtime.dll [External]
10:36:52:910 Loaded assembly: Java.Interop.dll [External]
10:36:52:911 Loaded assembly: System.Collections.dll [External]
10:36:52:911 Resolved pending breakpoint for 'Android.Runtime.JNIEnv.RegisterJniNatives(System.IntPtr, System.Int32, System.IntPtr, System.IntPtr, System.Int32)' to [0x0](no debug symbols).
10:36:52:911 Loaded assembly: System.Threading.dll [External]
10:36:52:911 Loaded assembly: System.Runtime.InteropServices.dll [External]
10:36:52:911 Loaded assembly: System.Threading.Thread.dll [External]
10:36:52:911 Loaded assembly: System.Diagnostics.StackTrace.dll [External]
10:36:52:914 [tance.mobile.ui] Attempt to remove non-JNI local reference, dumping thread
10:36:53:290 Loaded assembly: data-0x7c0d2596f050 [External]
10:36:53:313 Loaded assembly: data-0x7c0d2598f070 [External]
10:36:53:532 Loaded assembly: data-0x7c0dd1198a60 [External]
10:36:53:672 Loaded assembly: netstandard.dll [External]
10:36:53:731 Loaded assembly: System.Linq.dll [External]
10:36:53:766 [monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.VisualStudio.DesignTools.TapContract.dll
10:36:53:766 [monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.VisualStudio.DesignTools.TapContract.dll
10:36:53:780 [monodroid-assembly] open_from_bundles: failed to load assembly Xamarin.HotReload.Contracts.dll
10:36:53:780 [monodroid-assembly] open_from_bundles: failed to load assembly Xamarin.HotReload.Contracts.dll
10:36:53:801 Loaded assembly: System.Collections.Concurrent.dll [External]
10:36:53:974 Resolved pending breakpoint for 'Xamarin.HotReload.HotReloadAgent.BreakpointSendToIde(System.String)' to D:\a_work\1\s\HotReload\Source\Xamarin.HotReload.Agent\HotReloadAgent.cs:423 [0x00000].
10:36:53:974 Loaded assembly: System.Text.Json.dll [External]
10:36:53:974 Loaded assembly: System.Text.Encodings.Web.dll [External]
10:36:54:026 Loaded assembly: System.ComponentModel.dll [External]
10:36:54:816 [tance.mobile.ui] Symbol file data-0x7c0dd3e1c480 doesn't match image Microsoft.Maui.Controls.dll
10:36:54:816 [monodroid-assembly] open_from_bundles: failed to load assembly Xamarin.HotReload.Contracts.dll
10:36:54:817 [monodroid-assembly] open_from_bundles: failed to load assembly Xamarin.HotReload.Contracts.dll
10:36:54:852 Loaded assembly: Microsoft.Maui.Controls.dll [External]
10:36:54:852 [monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.VisualStudio.DesignTools.TapContract.dll
10:36:54:852 [monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.VisualStudio.DesignTools.TapContract.dll
10:36:54:910 Loaded assembly: System.ObjectModel.dll [External]
10:36:54:955 Loaded assembly: Microsoft.Maui.dll [External]
10:36:55:008 Loaded assembly: Microsoft.Maui.Graphics.dll [External]
10:36:55:089 Loaded assembly: Microsoft.Maui.Essentials.dll [External]
10:36:55:089 Loaded assembly: Microsoft.Extensions.Logging.Abstractions.dll [External]
10:36:55:102 Loaded assembly: System.ComponentModel.TypeConverter.dll [External]
10:36:55:142 Loaded assembly: System.Private.Uri.dll [External]
10:36:55:201 Loaded assembly: System.Runtime.Loader.dll [External]
10:36:55:275 Resolved pending breakpoint for 'Xamarin.HotReload.HotReloadAgent.BreakpointCheckpoint()' to D:\a_work\1\s\HotReload\Source\Xamarin.HotReload.Agent\HotReloadAgent.cs:418 [0x00000].
10:36:55:403 Loaded assembly: System.Memory.dll [External]
10:36:55:425 Loaded assembly: System.Reflection.Emit.Lightweight.dll [External]
10:36:55:471 Loaded assembly: System.Reflection.Emit.ILGeneration.dll [External]
10:36:55:497 Loaded assembly: System.Reflection.Primitives.dll [External]
10:36:55:508 Loaded assembly: System.Runtime.CompilerServices.Unsafe.dll [External]
10:36:55:712 Loaded assembly: Xamarin.AndroidX.Core.dll [External]
10:36:55:819 Loaded assembly: Microsoft.Extensions.DependencyInjection.Abstractions.dll [External]
10:36:55:939 Resolved pending breakpoint at 'MauiProgram.cs:24,1' to void Remittance.Mobile.UI.MauiProgram..cctor () [0x00000].
10:36:56:283 Resolved pending breakpoint at 'MauiProgram.cs:25,1' to void Remittance.Mobile.UI.MauiProgram..cctor () [0x00022].
10:36:56:313 Resolved pending breakpoint at 'MauiProgram.cs:46,1' to Microsoft.Maui.Hosting.MauiApp Remittance.Mobile.UI.MauiProgram.CreateMauiApp () [0x00087].
10:36:56:313 Resolved pending breakpoint at 'MauiProgram.cs:47,1' to Microsoft.Maui.Hosting.MauiApp Remittance.Mobile.UI.MauiProgram.CreateMauiApp () [0x000b2].
10:36:56:313 Resolved pending breakpoint at 'MauiProgram.cs:72,1' to Microsoft.Maui.Hosting.MauiApp Remittance.Mobile.UI.MauiProgram.CreateMauiApp () [0x00155].
10:36:56:313 Resolved pending breakpoint at 'MauiProgram.cs:74,1' to Microsoft.Maui.Hosting.MauiApp Remittance.Mobile.UI.MauiProgram.CreateMauiApp () [0x00167].
10:36:56:467 Resolved pending breakpoint at 'MauiProgram.cs:78,1' to Microsoft.Maui.Hosting.MauiApp Remittance.Mobile.UI.MauiProgram.CreateMauiApp () [0x00192].
10:37:03:441 Loaded assembly: Microsoft.Extensions.DependencyInjection.dll [External]
10:37:03:447 Loaded assembly: Microsoft.Maui.Controls.Xaml.dll [External]
10:37:03:466 Loaded assembly: System.Net.Http.dll [External]
10:37:03:486 Loaded assembly: Microsoft.AspNetCore.Components.Authorization.dll [External]
10:37:03:506 Loaded assembly: MudBlazor.dll [External]
10:37:03:525 Loaded assembly: Microsoft.Fast.Components.FluentUI.dll [External]
10:37:03:544 Loaded assembly: Microsoft.Extensions.Configuration.dll [External]
10:37:03:566 Loaded assembly: Microsoft.Extensions.Configuration.Abstractions.dll [External]
10:37:03:588 Loaded assembly: Microsoft.Extensions.Logging.dll [External]
10:37:03:610 Loaded assembly: Microsoft.AspNetCore.Components.WebView.Maui.dll [External]
10:37:03:632 Loaded assembly: Blazored.SessionStorage.dll [External]
10:37:03:662 Loaded assembly: Blazored.LocalStorage.dll [External]
10:37:03:682 Loaded assembly: Microsoft.Extensions.Http.dll [External]
10:37:03:704 Loaded assembly: Microsoft.AspNetCore.Authorization.dll [External]
10:37:03:725 Loaded assembly: Remittance.Shared.Core.dll
10:37:03:853 Loaded assembly: Xamarin.AndroidX.AppCompat.dll [External]
10:37:03:887 Loaded assembly: Xamarin.Google.Android.Material.dll [External]
10:37:03:912 Loaded assembly: Xamarin.AndroidX.SwipeRefreshLayout.dll [External]
10:37:04:189 Loaded assembly: Xamarin.AndroidX.Fragment.dll [External]
10:37:04:189 Loaded assembly: Xamarin.AndroidX.Activity.dll [External]
10:37:04:264 Loaded assembly: Xamarin.AndroidX.Lifecycle.ViewModel.dll [External]
10:37:04:298 Loaded assembly: Xamarin.AndroidX.Lifecycle.Common.dll [External]
10:37:04:314 Loaded assembly: Xamarin.AndroidX.SavedState.dll [External]
10:37:04:360 Loaded assembly: Xamarin.AndroidX.DrawerLayout.dll [External]
10:37:04:595 Loaded assembly: Xamarin.AndroidX.CustomView.dll [External]
10:37:04:596 Loaded assembly: Xamarin.AndroidX.Navigation.Runtime.dll [External]
10:37:04:614 Loaded assembly: System.Net.Primitives.dll [External]
10:37:04:652 Loaded assembly: Microsoft.Extensions.Options.dll [External]
10:37:04:662 Loaded assembly: Microsoft.JSInterop.dll [External]
10:37:04:690 [tance.mobile.ui] Symbol file data-0x7c0d159854b0 doesn't match image Microsoft.AspNetCore.Components.dll
10:37:04:690 Loaded assembly: Microsoft.AspNetCore.Components.dll [External]
10:37:04:697 [tance.mobile.ui] Symbol file data-0x7c0d159854b0 doesn't match image Microsoft.AspNetCore.Components.WebView.dll
10:37:04:716 Loaded assembly: Microsoft.AspNetCore.Components.WebView.dll [External]
10:37:04:736 Loaded assembly: Microsoft.AspNetCore.Components.Web.dll [External]
10:37:04:762 Loaded assembly: System.Diagnostics.Tracing.dll [External]
10:37:16:516 Loaded assembly: System.IdentityModel.Tokens.Jwt.dll [External]
10:37:16:549 Loaded assembly: Microsoft.IdentityModel.Tokens.dll [External]
10:37:26:865 Loaded assembly: Microsoft.Extensions.Primitives.dll [External]
10:37:26:882 Loaded assembly: AutoMapper.dll [External]
10:37:26:906 Loaded assembly: FluentValidation.dll [External]
10:37:26:927 Loaded assembly: Remittance.ApiClient.dll
10:37:26:945 Loaded assembly: FluentValidation.DependencyInjectionExtensions.dll [External]
10:37:26:968 Loaded assembly: System.Linq.Expressions.dll [External]
10:37:27:016 Loaded assembly: System.Text.RegularExpressions.dll [External]
10:37:28:150 [tance.mobile.ui] Explicit concurrent copying GC freed 40(18KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 1956KB/3913KB, paused 593us,58us total 4.604ms
10:37:28:586 Loaded assembly: System.Collections.Specialized.dll [External]
10:37:28:821 Loaded assembly: System.Reflection.Emit.dll [External]
10:37:37:228 Loaded assembly: System.Xml.ReaderWriter.dll [External]
10:37:37:249 Loaded assembly: System.Private.Xml.dll [External]
10:37:37:284 Loaded assembly: System.Text.Encoding.Extensions.dll [External]
10:37:37:474 Loaded assembly: System.Numerics.Vectors.dll [External]
10:37:37:628 Thread started: https://github.com/dotnet/maui/pull/2
10:37:37:647 Thread started: https://github.com/dotnet/maui/issues/3
10:37:37:662 Thread started: https://github.com/dotnet/maui/pull/4
10:37:37:750 Loaded assembly: System.Runtime.Serialization.Json.dll [External]
10:37:37:770 Loaded assembly: System.Private.DataContractSerialization.dll [External]
10:37:37:790 Loaded assembly: System.Runtime.Serialization.Xml.dll [External]
10:37:37:842 Loaded assembly: System.Runtime.Serialization.Primitives.dll [External]
10:37:38:078 Loaded assembly: Microsoft.Extensions.FileProviders.Abstractions.dll [External]
10:37:38:207 Loaded assembly: Xamarin.AndroidX.RecyclerView.dll [External]
10:37:38:221 Loaded assembly: Xamarin.AndroidX.CardView.dll [External]
10:37:38:473 [libEGL] loaded /vendor/lib64/egl/libEGL_emulation.so
10:37:38:473 [libEGL] loaded /vendor/lib64/egl/libGLESv1_CM_emulation.so
10:37:38:483 [libEGL] loaded /vendor/lib64/egl/libGLESv2_emulation.so
10:37:38:501 Loaded assembly: Xamarin.AndroidX.Loader.dll [External]
10:37:53:010 Loaded assembly: Microsoft.Maui.Controls.Compatibility.dll [External]
10:37:53:104 Loaded assembly: Xamarin.AndroidX.CoordinatorLayout.dll [External]
10:37:53:227 [WebViewFactory] Loading com.google.android.webview version 103.0.5060.71 (code 506007137)
10:37:53:239 [nativeloader] Configuring classloader-namespace for other apk /data/app/~~zZ2fX-2BQwo0u-3WNx9bMg==/com.google.android.trichromelibrary_506007137-ilNom3WZlC9AFcHkG0h5SA==/TrichromeLibrary.apk. target_sdk_version=33, uses_libraries=ALL, library_path=/data/app/~~fhFD3hL-LdBGQGTWGJZLsg==/com.google.android.webview-tTxDeyGMY7vXis8ci5xGig==/lib/x86_64:/data/app/~~fhFD3hL-LdBGQGTWGJZLsg==/com.google.android.webview-tTxDeyGMY7vXis8ci5xGig==/WebViewGoogle.apk!/lib/x86_64:/data/app/~~zZ2fX-2BQwo0u-3WNx9bMg==/com.google.android.trichromelibrary_506007137-ilNom3WZlC9AFcHkG0h5SA==/TrichromeLibrary.apk!/lib/x86_64, permitted_path=/data:/mnt/expand
10:37:53:260 [nativeloader] Configuring classloader-namespace for other apk /data/app/~~fhFD3hL-LdBGQGTWGJZLsg==/com.google.android.webview-tTxDeyGMY7vXis8ci5xGig==/WebViewGoogle.apk. target_sdk_version=33, uses_libraries=, library_path=/data/app/~~fhFD3hL-LdBGQGTWGJZLsg==/com.google.android.webview-tTxDeyGMY7vXis8ci5xGig==/lib/x86_64:/data/app/~~fhFD3hL-LdBGQGTWGJZLsg==/com.google.android.webview-tTxDeyGMY7vXis8ci5xGig==/WebViewGoogle.apk!/lib/x86_64:/data/app/~~zZ2fX-2BQwo0u-3WNx9bMg==/com.google.android.trichromelibrary_506007137-ilNom3WZlC9AFcHkG0h5SA==/TrichromeLibrary.apk!/lib/x86_64, permitted_path=/data:/mnt/expand
10:37:53:637 [tance.mobile.ui] Accessing hidden method Landroid/os/Trace;->isTagEnabled(J)Z (unsupported, reflection, allowed)
10:37:53:637 [tance.mobile.ui] Accessing hidden method Landroid/os/Trace;->traceBegin(JLjava/lang/String;)V (unsupported, reflection, allowed)
10:37:53:651 [tance.mobile.ui] Accessing hidden method Landroid/os/Trace;->traceEnd(J)V (unsupported, reflection, allowed)
10:37:53:654 [tance.mobile.ui] Accessing hidden method Landroid/os/Trace;->asyncTraceBegin(JLjava/lang/String;I)V (unsupported, reflection, allowed)
10:37:53:654 [tance.mobile.ui] Accessing hidden method Landroid/os/Trace;->asyncTraceEnd(JLjava/lang/String;I)V (unsupported, reflection, allowed)
10:37:53:671 [cr_WVCFactoryProvider] Loaded version=103.0.5060.71 minSdkVersion=29 isBundle=false multiprocess=true packageId=2
10:37:53:728 [cr_VariationsUtils] Failed reading seed file "/data/user/0/com.companyname.remittance.mobile.ui/app_webview/variations_seed_new"
10:37:53:728 [cr_VariationsUtils] Failed reading seed file "/data/user/0/com.companyname.remittance.mobile.ui/app_webview/variations_seed"
10:37:53:736 [cr_VariationsUtils] Requesting new seed from IVariationsSeedServer
10:37:53:807 [cr_LibraryLoader] Successfully loaded native library
10:37:53:813 [cr_CachingUmaRecorder] Flushed 9 samples from 9 histograms.
10:37:54:481 [CompatibilityChangeReporter] Compat change id reported: 183155436; UID 10164; state: DISABLED
10:37:55:093 [CompatibilityChangeReporter] Compat change id reported: 214741472; UID 10164; state: DISABLED
10:37:55:161 [CompatibilityChangeReporter] Compat change id reported: 171228096; UID 10164; state: ENABLED
10:37:55:521 Loaded assembly: Microsoft.Extensions.FileProviders.Embedded.dll [External]
10:37:55:521 Loaded assembly: Microsoft.Extensions.FileProviders.Composite.dll [External]
10:37:55:617 Loaded assembly: System.Xml.XDocument.dll [External]
10:37:55:648 Loaded assembly: System.Private.Xml.Linq.dll [External]
10:37:56:523 [tance.mobile.ui] Accessing hidden method Landroid/media/AudioManager;->getOutputLatency(I)I (unsupported, reflection, allowed)
10:37:56:523 [HostConnection] createUnique: call
10:37:56:523 [HostConnection] HostConnection::get() New Host Connection established 0x7c0bb5941850, tid 20655
10:37:56:523 [HostConnection] HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_vulkan_queue_submit_with_commands ANDROID_EMU_sync_buffer_data ANDROID_EMU_read_color_buffer_dma ANDROID_EMU_hwc_multi_configs GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_3_1
10:37:56:523 [EGL_emulation] eglCreateContext: 0x7c0bb5941b50: maj 3 min 1 rcv 4
10:37:56:523 [EGL_emulation] eglMakeCurrent: 0x7c0bb5941b50: ver 3 1 (tinfo 0x7c0dd0926080) (first time)
10:37:56:523 [cr_media] Requires BLUETOOTH permission
10:37:56:523 [tance.mobile.ui] Background young concurrent copying GC freed 7015(971KB) AllocSpace objects, 3(124KB) LOS objects, 29% free, 2747KB/3913KB, paused 7.731ms,615us total 60.144ms
10:37:56:558 Loaded assembly: Xamarin.AndroidX.Lifecycle.LiveData.Core.dll [External]
10:37:56:596 [tance.mobile.ui] Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (unsupported, reflection, allowed)
10:37:56:603 [tance.mobile.ui] Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (unsupported, reflection, allowed)
10:38:22:939 Thread started: https://github.com/dotnet/maui/pull/5
10:38:22:948 [EGL_emulation] eglCreateContext: 0x7c0bb5941b50: maj 3 min 0 rcv 3
10:38:23:077 Thread started: .NET Timers https://github.com/dotnet/maui/issues/6
10:38:23:176 [Choreographer] Skipped 1636 frames! The application may be doing too much work on its main thread.
10:38:23:201 [EGL_emulation] eglCreateContext: 0x7c0bb5941490: maj 3 min 0 rcv 3
10:38:23:238 [CompatibilityChangeReporter] Compat change id reported: 193247900; UID 10164; state: DISABLED
10:38:23:250 [tance.mobile.ui] Background concurrent copying GC freed 5717(842KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 3270KB/6541KB, paused 2.477ms,100us total 114.162ms
10:38:23:349 [HostConnection] createUnique: call
10:38:23:349 [HostConnection] HostConnection::get() New Host Connection established 0x7c0bb5942690, tid 20527
10:38:23:359 [HostConnection] HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_vulkan_queue_submit_with_commands ANDROID_EMU_sync_buffer_data ANDROID_EMU_read_color_buffer_dma ANDROID_EMU_hwc_multi_configs GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_3_1
10:38:23:374 [OpenGLRenderer] Failed to choose config with EGL_SWAP_BEHAVIOR_PRESERVED, retrying without...
10:38:23:411 [OpenGLRenderer] Failed to initialize 101010-2 format, error = EGL_SUCCESS
10:38:23:467 [EGL_emulation] eglCreateContext: 0x7c0bb59437d0: maj 3 min 1 rcv 4
10:38:23:467 [EGL_emulation] eglMakeCurrent: 0x7c0bb59437d0: ver 3 1 (tinfo 0x7c0dd0926100) (first time)
10:38:23:794 [Gralloc4] mapper 4.x is not supported
10:38:23:794 [HostConnection] createUnique: call
10:38:23:794 [HostConnection] HostConnection::get() New Host Connection established 0x7c0bb59428d0, tid 20527
10:38:23:794 [goldfish-address-space] allocate: Ask for block of size 0x100
10:38:23:794 [goldfish-address-space] allocate: ioctl allocate returned offset 0x3f3ffe000 size 0x2000
10:38:23:794 [Gralloc4] allocator 4.x is not supported
10:38:23:966 [HostConnection] HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_vulkan_queue_submit_with_commands ANDROID_EMU_sync_buffer_data ANDROID_EMU_read_color_buffer_dma ANDROID_EMU_hwc_multi_configs GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_3_1
10:38:24:546 [Parcel] Expecting binder but got null!
10:38:24:549 [OpenGLRenderer] Davey! duration=28205ms; Flags=1, FrameTimelineVsyncId=30168, IntendedVsync=9003218538679, Vsync=9030485204255, InputEventId=0, HandleInputStart=9030495063302, AnimationStart=9030495105902, PerformTraversalsStart=9030497092302, DrawStart=9031150740902, FrameDeadline=9003235205345, FrameInterval=9030494620902, FrameStartTime=16666666, SyncQueued=9031239418402, SyncStart=9031283691502, IssueDrawCommandsStart=9031286004102, SwapBuffers=9031403663802, FrameCompleted=9031468370102, DequeueBufferDuration=39900, QueueBufferDuration=1691400, GpuCompleted=9031467568302, SwapBuffersCompleted=9031468370102, DisplayPresentTime=-4294967296, CommandSubmissionCompleted=9031403663802,
10:38:24:559 [Choreographer] Skipped 68 frames! The application may be doing too much work on its main thread.
10:38:24:560 [OpenGLRenderer] Davey! duration=1205ms; Flags=0, FrameTimelineVsyncId=30215, IntendedVsync=9030501870921, Vsync=9031635204209, InputEventId=0, HandleInputStart=9031648029702, AnimationStart=9031648073102, PerformTraversalsStart=9031670113702, DrawStart=9031674061102, FrameDeadline=9031485204215, FrameInterval=9031647652002, FrameStartTime=16666666, SyncQueued=9031700040702, SyncStart=9031700573402, IssueDrawCommandsStart=9031700975302, SwapBuffers=9031702513102, FrameCompleted=9031708053302, DequeueBufferDuration=513400, QueueBufferDuration=2492200, GpuCompleted=9031706217202, SwapBuffersCompleted=9031708053302, DisplayPresentTime=8589934600, CommandSubmissionCompleted=9031702513102,
10:38:25:115 Thread started: https://github.com/dotnet/maui/issues/7
10:38:25:140 Thread started: https://github.com/dotnet/maui/issues/8
10:38:25:141 Thread started: https://github.com/dotnet/maui/issues/9
10:38:25:203 Thread started: https://github.com/dotnet/maui/issues/10
10:38:25:203 Thread started: https://github.com/dotnet/maui/issues/11
10:38:25:787 Thread started: https://github.com/dotnet/maui/issues/12
10:38:25:810 Thread started: https://github.com/dotnet/maui/issues/13
10:38:26:030 Thread started: https://github.com/dotnet/maui/issues/14
10:38:26:032 Thread started: https://github.com/dotnet/maui/issues/15
10:38:26:033 Thread started: https://github.com/dotnet/maui/issues/16
10:38:26:153 Thread started: https://github.com/dotnet/maui/issues/17
10:38:26:153 Thread started: .NET ThreadPool Gate https://github.com/dotnet/maui/issues/18
10:38:26:491 Thread started: https://github.com/dotnet/maui/issues/19
10:38:26:491 Loaded assembly: System.Runtime.Intrinsics.dll [External]
10:38:28:192 Thread started: https://github.com/dotnet/maui/issues/20
10:38:28:203 Thread started: https://github.com/dotnet/maui/issues/21
10:38:35:790 Thread started: https://github.com/dotnet/maui/issues/22
10:38:35:832 [chromium] [INFO:CONSOLE(137)] "Uncaught ReferenceError: AOS is not defined", source: https://0.0.0.0/ (137)
10:38:35:837 [chromium] [INFO:CONSOLE(1)] "Uncaught ReferenceError: $ is not defined", source: https://0.0.0.0/_content/Remittance.Shared.UI/js/sidebarmenu.js (1)
10:38:35:844 [chromium] [INFO:CONSOLE(1)] "Uncaught ReferenceError: $ is not defined", source: https://0.0.0.0/_content/Remittance.Shared.UI/js/custom.js (1)
10:38:35:860 [chromium] [INFO:CONSOLE(7)] "Uncaught ReferenceError: $ is not defined", source: https://0.0.0.0/_content/Remittance.Shared.UI/js/sessiontimeout.js (7)
10:38:35:901 [EGL_emulation] app_time_stats: avg=893.55ms min=3.78ms max=10820.11ms count=13
10:38:36:112 Thread started: https://github.com/dotnet/maui/issues/23
10:38:36:320 [tance.mobile.ui] Explicit concurrent copying GC freed 20008(1152KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 3653KB/7306KB, paused 936us,141us total 19.488ms
10:38:36:663 Loaded assembly: Remittance.Shared.UI.dll
10:38:36:727 Loaded assembly: Microsoft.AspNetCore.Metadata.dll [External]
10:38:36:917 Loaded assembly: Anonymously Hosted DynamicMethods Assembly [External]
10:38:37:057 Loaded assembly: System.Security.Claims.dll [External]
10:38:37:344 Loaded assembly: Humanizer.dll [External]
10:38:37:427 [monodroid-assembly] open_from_bundles: failed to load assembly en-US/Humanizer.resources.dll
10:38:37:427 [monodroid-assembly] open_from_bundles: failed to load assembly en-US/Humanizer.resources.dll
10:38:37:613 [Choreographer] Skipped 82 frames! The application may be doing too much work on its main thread.
10:38:37:781 Loaded assembly: Newtonsoft.Json.dll [External]
10:38:37:894 Loaded assembly: System.Security.Cryptography.X509Certificates.dll [External]
10:38:37:906 Thread started: https://github.com/dotnet/maui/issues/24
10:38:38:023 Loaded assembly: System.Diagnostics.DiagnosticSource.dll [External]
10:38:38:279 Loaded assembly: System.Collections.NonGeneric.dll [External]
10:38:38:859 Loaded assembly: System.Net.Requests.dll [External]
10:38:39:073 [Choreographer] Skipped 80 frames! The application may be doing too much work on its main thread.
10:38:39:300 [EGL_emulation] app_time_stats: avg=1069.27ms min=21.87ms max=2960.40ms count=3
10:38:39:300 [OpenGLRenderer] Davey! duration=1395ms; Flags=0, FrameTimelineVsyncId=35134, IntendedVsync=9045035203673, Vsync=9046368536953, InputEventId=0, HandleInputStart=9046378639702, AnimationStart=9046379452102, PerformTraversalsStart=9046389583602, DrawStart=9046390173602, FrameDeadline=9045051870339, FrameInterval=9046376050402, FrameStartTime=16666666, SyncQueued=9046390749802, SyncStart=9046390980202, IssueDrawCommandsStart=9046391250002, SwapBuffers=9046424249602, FrameCompleted=9046431395502, DequeueBufferDuration=51000, QueueBufferDuration=2297100, GpuCompleted=9046430013902, SwapBuffersCompleted=9046431395502, DisplayPresentTime=-4294967294, CommandSubmissionCompleted=9046424249602,
10:38:40:087 System.NullReferenceException: 'Object reference not set to an instance of an object.'
10:38:40:087
10:38:40:162 Thread started: .NET ThreadPool Worker https://github.com/dotnet/maui/pull/25
10:39:25:800 System.NullReferenceException: 'Object reference not set to an instance of an object.'
10:39:25:800
10:39:25:800 The thread 0x15 has exited with code 0 (0x0).
10:39:25:800 The thread 0x19 has exited with code 0 (0x0).
10:39:25:895 The thread 0x11 has exited with code 0 (0x0).
10:39:25:909 Thread finished: https://github.com/dotnet/maui/issues/21
10:39:25:909 Thread finished: .NET ThreadPool Worker https://github.com/dotnet/maui/pull/25
10:39:25:911 Thread finished: https://github.com/dotnet/maui/issues/17
10:39:26:095 Thread started: https://github.com/dotnet/maui/issues/26
10:39:26:107 Thread started: https://github.com/dotnet/maui/issues/27
10:39:29:217 [mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object.
10:39:29:218 [mono-rt] at Microsoft.AspNetCore.Components.WebView.Services.WebViewJSRuntime.BeginInvokeJS(Int64 taskId, String identifier, String argsJson, JSCallResultType resultType, Int64 targetInstanceId)
10:39:29:218 [mono-rt] at Microsoft.JSInterop.JSRuntime.InvokeAsync[String](Int64 targetInstanceId, String identifier, CancellationToken cancellationToken, Object[] args)
10:39:29:218 [mono-rt] at Microsoft.JSInterop.JSRuntime.InvokeAsync[String](String identifier, CancellationToken cancellationToken, Object[] args)
10:39:29:218 [mono-rt] at Microsoft.JSInterop.JSRuntimeExtensions.InvokeAsync[String](IJSRuntime jsRuntime, String identifier, CancellationToken cancellationToken, Object[] args)
10:39:29:219 [mono-rt] at Blazored.SessionStorage.BrowserStorageProvider.GetItemAsync(String key, CancellationToken cancellationToken)
10:39:29:219 [mono-rt] at Remittance.Mobile.UI.MauiProgram.<>c__DisplayClass2_0.<b__2>d.MoveNext() in G:\SAM_Project\swap-money-remittance-CustomerPortal\Remittance.Mobile.UI\MauiProgram.cs:line 51
10:39:29:219 [mono-rt] --- End of stack trace from previous location ---
10:39:29:219 [mono-rt] at System.Threading.Tasks.Task.<>c.b__128_0(Object state)
10:39:29:219 [mono-rt] at Android.App.SyncContext.<>c__DisplayClass2_0.b__0()
10:39:29:219 [mono-rt] at Java.Lang.Thread.RunnableImplementor.Run()
10:39:29:219 [mono-rt] at Java.Lang.IRunnableInvoker.n_Run(IntPtr jnienv, IntPtr native__this)
10:39:29:219 [mono-rt] at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz)
10:39:29:224 [HostConnection] createUnique: call
10:39:29:224 [HostConnection] HostConnection::get() New Host Connection established 0x7c0bb59443d0, tid 20277
10:39:29:228 [HostConnection] HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_vulkan_queue_submit_with_commands ANDROID_EMU_sync_buffer_data ANDROID_EMU_read_color_buffer_dma ANDROID_EMU_hwc_multi_configs GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_3_1
10:39:29:231 [libEGL] call to OpenGL ES API with no current context (logged once per thread)
10:39:29:236 [libEGL] validate_display:537 error 3008 (EGL_BAD_DISPLAY)
10:39:29:236 [libEGL] validate_display:537 error 3008 (EGL_BAD_DISPLAY)
10:39:29:236 [libEGL] validate_display:537 error 3008 (EGL_BAD_DISPLAY)
10:39:29:236 [libEGL] validate_display:537 error 3008 (EGL_BAD_DISPLAY)
10:39:29:755 [BpBinder] onLastStrongRef automatically unlinking death recipients:
12:29:24:365 Starting Android application ...
12:29:24:750 > am start -a "android.intent.action.MAIN" -c "android.intent.category.LAUNCHER" -n "com.companyname.remittance.mobile.ui/crc64a20e96d7d12de042.MainActivity"
12:29:25:016 > Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=com.companyname.remittance.mobile.ui/crc64a20e96d7d12de042.MainActivity }
12:29:25:016

--- a/src/BlazorWebView/samples/MauiRazorClassLibrarySample/wwwroot/MauiRazorClassLibrarySample.lib.module.js
+++ b/src/BlazorWebView/samples/MauiRazorClassLibrarySample/wwwroot/MauiRazorClassLibrarySample.lib.module.js
@@ -1,0 +1,3 @@
+﻿const element = document.createElement('p');
+element.innerHTML = 'Hello from Razor Class Library';
+document.body.appendChild(element);

--- a/src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs
+++ b/src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs
@@ -1,4 +1,5 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Runtime.InteropServices.WindowsRuntime;
 using System.Threading.Tasks;
@@ -84,86 +85,113 @@ namespace Microsoft.AspNetCore.Components.WebView.Maui
 
 			_logger.HandlingWebRequest(requestUri);
 
-			// First, call into WebViewManager to see if it has a framework file for this request. It will
-			// fall back to an IFileProvider, but on WinUI it's always a NullFileProvider, so that will never
-			// return a file.
-			if (TryGetResponseContent(requestUri, allowFallbackOnHostPage, out var statusCode, out var statusMessage, out var content, out var headers)
+			var uri = new Uri(requestUri);
+			var relativePath = AppOriginUri.IsBaseOf(uri) ? AppOriginUri.MakeRelativeUri(uri).ToString() : null;
+
+			// Check if the uri is _framework/blazor.modules.json is a special case as the built-in file provider
+			// brings in a default implementation.
+			if (relativePath != null &&
+				string.Equals(relativePath, "_framework/blazor.modules.json", StringComparison.Ordinal) &&
+				await TryServeFromFolderAsync(eventArgs, allowFallbackOnHostPage: false, requestUri, relativePath))
+			{
+				_logger.ResponseContentBeingSent(requestUri, 200);
+			}
+			else if (TryGetResponseContent(requestUri, allowFallbackOnHostPage, out var statusCode, out var statusMessage, out var content, out var headers)
 				&& statusCode != 404)
 			{
+				// First, call into WebViewManager to see if it has a framework file for this request. It will
+				// fall back to an IFileProvider, but on WinUI it's always a NullFileProvider, so that will never
+				// return a file.
 				var headerString = GetHeaderString(headers);
 				_logger.ResponseContentBeingSent(requestUri, statusCode);
 				eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(content.AsRandomAccessStream(), statusCode, statusMessage, headerString);
 			}
-			else if (new Uri(requestUri) is Uri uri && AppOriginUri.IsBaseOf(uri))
+			else if (relativePath != null)
 			{
-				var relativePath = AppOriginUri.MakeRelativeUri(uri).ToString();
-
-				// If the path does not end in a file extension (or is empty), it's most likely referring to a page,
-				// in which case we should allow falling back on the host page.
-				if (allowFallbackOnHostPage && !Path.HasExtension(relativePath))
-				{
-					relativePath = _hostPageRelativePath;
-				}
-				relativePath = Path.Combine(_contentRootRelativeToAppRoot, relativePath.Replace('/', '\\'));
-				statusCode = 200;
-				statusMessage = "OK";
-				var contentType = StaticContentProvider.GetResponseContentTypeOrDefault(relativePath);
-				headers = StaticContentProvider.GetResponseHeaders(contentType);
-				IRandomAccessStream? stream = null;
-				if (_isPackagedApp)
-				{
-					var winUIItem = await Package.Current.InstalledLocation.TryGetItemAsync(relativePath);
-					if (winUIItem != null)
-					{
-						using var contentStream = await Package.Current.InstalledLocation.OpenStreamForReadAsync(relativePath);
-						stream = await CopyContentToRandomAccessStreamAsync(contentStream);
-					}
-				}
-				else
-				{
-					var path = Path.Combine(AppContext.BaseDirectory, relativePath);
-					if (File.Exists(path))
-					{
-						using var contentStream = File.OpenRead(path);
-						stream = await CopyContentToRandomAccessStreamAsync(contentStream);
-					}
-				}
-
-				var hotReloadedContent = Stream.Null;
-				if (StaticContentHotReloadManager.TryReplaceResponseContent(_contentRootRelativeToAppRoot, requestUri, ref statusCode, ref hotReloadedContent, headers))
-				{
-					stream = await CopyContentToRandomAccessStreamAsync(hotReloadedContent);
-				}
-
-				if (stream != null)
-				{
-					var headerString = GetHeaderString(headers);
-
-					_logger.ResponseContentBeingSent(requestUri, statusCode);
-
-					eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(
-						stream,
-						statusCode,
-						statusMessage,
-						headerString);
-				}
-				else
-				{
-					_logger.ReponseContentNotFound(requestUri);
-				}
-
-				async Task<IRandomAccessStream> CopyContentToRandomAccessStreamAsync(Stream content)
-				{
-					using var memStream = new MemoryStream();
-					await content.CopyToAsync(memStream);
-					var randomAccessStream = new InMemoryRandomAccessStream();
-					await randomAccessStream.WriteAsync(memStream.GetWindowsRuntimeBuffer());
-					return randomAccessStream;
-				}
+				await TryServeFromFolderAsync(
+					eventArgs,
+					allowFallbackOnHostPage,
+					requestUri,
+					relativePath);
 			}
 
 			// Notify WebView2 that the deferred (async) operation is complete and we set a response.
 			deferral.Complete();
+		}
+
+		private async Task<bool> TryServeFromFolderAsync(
+			CoreWebView2WebResourceRequestedEventArgs eventArgs,
+			bool allowFallbackOnHostPage,
+			string requestUri,
+			string relativePath)
+		{
+			// If the path does not end in a file extension (or is empty), it's most likely referring to a page,
+			// in which case we should allow falling back on the host page.
+			if (allowFallbackOnHostPage && !Path.HasExtension(relativePath))
+			{
+				relativePath = _hostPageRelativePath;
+			}
+			relativePath = Path.Combine(_contentRootRelativeToAppRoot, relativePath.Replace('/', '\\'));
+			var statusCode = 200;
+			var statusMessage = "OK";
+			var contentType = StaticContentProvider.GetResponseContentTypeOrDefault(relativePath);
+			var headers = StaticContentProvider.GetResponseHeaders(contentType);
+			IRandomAccessStream? stream = null;
+			if (_isPackagedApp)
+			{
+				var winUIItem = await Package.Current.InstalledLocation.TryGetItemAsync(relativePath);
+				var location = Package.Current.InstalledLocation.Path;
+				if (winUIItem != null)
+				{
+					using var contentStream = await Package.Current.InstalledLocation.OpenStreamForReadAsync(relativePath);
+					stream = await CopyContentToRandomAccessStreamAsync(contentStream);
+				}
+			}
+			else
+			{
+				var path = Path.Combine(AppContext.BaseDirectory, relativePath);
+				if (File.Exists(path))
+				{
+					using var contentStream = File.OpenRead(path);
+					stream = await CopyContentToRandomAccessStreamAsync(contentStream);
+				}
+			}
+
+			var hotReloadedContent = Stream.Null;
+			if (StaticContentHotReloadManager.TryReplaceResponseContent(_contentRootRelativeToAppRoot, requestUri, ref statusCode, ref hotReloadedContent, headers))
+			{
+				stream = await CopyContentToRandomAccessStreamAsync(hotReloadedContent);
+			}
+
+			if (stream != null)
+			{
+				var headerString = GetHeaderString(headers);
+
+				_logger.ResponseContentBeingSent(requestUri, statusCode);
+
+				eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(
+					stream,
+					statusCode,
+					statusMessage,
+					headerString);
+
+				return true;
+			}
+			else
+			{
+				_logger.ReponseContentNotFound(requestUri);
+			}
+
+			return false;
+
+			async Task<IRandomAccessStream> CopyContentToRandomAccessStreamAsync(Stream content)
+			{
+				using var memStream = new MemoryStream();
+				await content.CopyToAsync(memStream);
+				var randomAccessStream = new InMemoryRandomAccessStream();
+				await randomAccessStream.WriteAsync(memStream.GetWindowsRuntimeBuffer());
+				return randomAccessStream;
+			}
 		}
 
 		/// <inheritdoc />

--- a/src/Controls/samples/Controls.Sample/wwwroot/Maui.Controls.Sample.lib.module.js
+++ b/src/Controls/samples/Controls.Sample/wwwroot/Maui.Controls.Sample.lib.module.js
@@ -1,0 +1,3 @@
+﻿const element = document.createElement('p');
+element.innerHTML = 'Hello from App';
+document.body.appendChild(element);

